File: Python/BOJ/18258 큐 2.py
import sys
from collections import deque

N = int(sys.stdin.readline())
queue = deque()
for _ in range(N):
    A = sys.stdin.readline().split()
    if A[0] == 'push':
        queue.append(A[1])
    elif A[0] == 'pop':
        if len(queue) == 0:
            print(-1)
        else :
            print(queue.popleft())
    elif A[0] == 'size':
        print(len(queue))
    elif A[0] == 'empty':
        if len(queue) == 0:
            print(1)
        else :
            print(0)
    elif A[0] == 'front':
        if len(queue) == 0 :
            print(-1)
        else:
            print(queue[0])
    elif A[0] == 'back':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[-1])

File: Java/BOJ/18258 큐2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        Deque<String> queue = new LinkedList<>();
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            List<String> x = new ArrayList<>();
            while (st.hasMoreTokens()) {
                x.add(st.nextToken());
            }
            if (x.get(0).equals("push")) {
                queue.add(x.get(1));
            } else if (x.get(0).equals("front")) {
                if (!queue.isEmpty()) {
                    sb.append(queue.peek()).append("\n");
                } else {
                	sb.append(-1).append("\n");
                }
            } else if (x.get(0).equals("back")) {
                if (!queue.isEmpty()) {
                    sb.append(queue.peekLast()).append("\n");
                } else {
                	sb.append(-1).append("\n");
                }
            } else if (x.get(0).equals("pop")) {
                if (!queue.isEmpty()) {
                    sb.append(queue.poll()).append("\n");
                } else {
                	sb.append(-1).append("\n");
                }
            } else if (x.get(0).equals("size")) {
                sb.append(queue.size()).append("\n");
            } else if (x.get(0).equals("empty")) {
                if (queue.isEmpty()) {
                	sb.append(1).append("\n");
                } else {
                	sb.append(0).append("\n");
                }
            }
        }
        System.out.println(sb);
    }
}